compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -330,11 +330,9 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
         // equate the two trait refs after normalization
         let a = ocx.normalize(&cause, self.param_env, a);
         let b = ocx.normalize(&cause, self.param_env, b);
-        if ocx.eq(&cause, self.param_env, a, b).is_ok() {
-            if ocx.select_all_or_error().is_empty() {
-                // All good.
-                return true;
-            }
+        if ocx.eq(&cause, self.param_env, a, b).is_ok() && ocx.select_all_or_error().is_empty() {
+            // All good.
+            return true;
         }
         return false;
```

Comment:
I mean at this point we should just make this return `ocx.eq(&cause, self.param_env, a, b).is_ok() && ocx.select_all_or_error().is_empty()`, I think. I was using two `if` to make the sequencing more explicit.

Reply:
right, I missed the trivial body here.

For what it's worth, I actually tend to write this like two separate if statements returning false since those are the "exceptional" path.

Reply:
Feel free to change this to the usual style used with infctx, I don't usually interact much with that type so I don't know the conventions.^^